qiskit/transpiler/passes/synthesis/high_level_synthesis.py

```diff
@@ -644,3 +647,80 @@ def run(self, high_level_object, coupling_map=None, target=None, qubits=None, **
         """Run synthesis for the given Permutation."""
         decomposition = synth_permutation_acg(high_level_object.pattern)
         return decomposition
+
+
+class TokenSwapperSynthesisPermutation(HighLevelSynthesisPlugin):
```

Comment:
We need to add this class somewhere in the toctree to render the docstring here. Right now this class (nor any of the HLS plugins) is not being included in the documentation builds. I was talking to @Cryoris offline about this a bit the other day as there isn't a unified place to document this right now things are spread out a bit too much in the organizational structure. I think for right now if you added a  HLS Plugins section to the module docstring in `qiskit/transpiler/passes/synthesis/plugin.py` and added an autosummary for this class that'd be enough. We can refactor the organizational structure in a follow up easily enough.